runtime/compiler/optimizer/UnsafeFastPath.cpp

```diff
@@ -560,7 +560,7 @@ int32_t TR_UnsafeFastPath::perform()
                else
                   {
                   //This is a load
-                  if (isByIndex && node->getDataType() != type)
+                  if (node->getDataType() != type)
```

Comment:
There seems to be an assert below guarding for future modifications. Are you certain we won't hit this assert? Note it's not a `FATAL` assert so it won't fire by default (outside of debug builds).

Reply:
Not for Unsafe methods. Is there similar problem to JITHelpers methods like `getByteFromArray`? The method return `byte` but in trees we use `icallx`, so a conversion is needed on the load.

Reply:
Now that you mention, I'll update the assertion message to not limit to JITHelpers.

Reply:
@fjeremic Have my comments answered your questions? Any more concerns?